File: Python/007__Reverse_Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        result = 0
        if x > 0 :
            while x != 0:
                result = result * 10 + x % 10
                x /= 10
        else :
            x *= -1
            while x != 0:
                result = result * 10 + x % 10
                x /= 10
            result *= -1
        if result > 2 ** 31 or result < - 2 ** 31:
            return 0

        return result

File: Java/007__Reverse_Integer.java
class Solution {
    public int reverse(int x) {
        int result = 0;
        while (x != 0) {
            int remain = x % 10;
            x /= 10;
            if (result > Integer.MAX_VALUE/10 || (result == Integer.MAX_VALUE / 10 && remain > 7)) {
                return 0;
            }
            if (result < Integer.MIN_VALUE/10 || (result == Integer.MIN_VALUE / 10 && remain < -8)) {
                return 0;
            }
            result = result * 10 + remain;
        }
        return result;
    }
}
